save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/AwesomeBenchmarksView.kt

```diff
@@ -514,3 +525,51 @@ class AwesomeBenchmarksView : AbstractView<PropsWithChildren, AwesomeBenchmarksS
         }
     }
 }
+
+/**
+ * The function of analyzing the URL of a tabbed page goes to the tab that was entered in the url, according to the role
+ *
+ * @param menu
+ * @param role
+ * @param flag
+ */
+fun <T, S> CComponent<*, S>.urlAnalysis(menu: MenuBar<T>, role: Role, flag: Boolean?)
```

Comment:
what is `MenuBar` - I see this name in ContestView and don't understand, why I see it here

Reply:
Renamed ```MenuBar``` to ```TabMenuBar```